File: frontend/src/types/workflow.ts
// ============================================================
// Company OS — Workflow Types
// ============================================================

export interface WorkflowStage {
  id: string;
  workflow_id: string;
  name: string;
  code: string;
  stage_type: string;
  order_index: number;
  default_folder_name?: string;
  created_at: string;
}

export interface WorkflowStageCreate {
  name: string;
  code: string;
  stage_type?: string;
  order_index?: number;
  default_folder_name?: string;
}

export interface WorkflowTransition {
  id: string;
  workflow_id: string;
  from_stage_id: string;
  to_stage_id: string;
  from_stage_name?: string;
  to_stage_name?: string;
  action: string;
  required_permission?: string;
  required_fields?: string[];
  auto_create_task_type_id?: string;
  task_type_name?: string;
  assignment_mode: string;
  active: boolean;
  created_at: string;
}

export interface WorkflowTransitionCreate {
  from_stage_id: string;
  to_stage_id: string;
  action: string;
  required_permission?: string;
  required_fields?: string[];
  auto_create_task_type_id?: string;
  assignment_mode?: string;
  active?: boolean;
}

export interface Workflow {
  id: string;
  name: string;
  code: string;
  description?: string;
  active: boolean;
  created_at: string;
  updated_at: string;
  stages: WorkflowStage[];
  transitions: WorkflowTransition[];
}

export interface WorkflowCreate {
  name: string;
  code: string;
  description?: string;
  active?: boolean;
}

File: app/schemas/workflow.py
import uuid
from typing import Optional, List, Any
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class WorkflowStageBase(BaseModel):
    name: str
    code: str
    stage_type: str = "standard"
    order_index: int = 0
    default_folder_name: Optional[str] = None


class WorkflowStageCreate(WorkflowStageBase):
    pass


class WorkflowStageOut(WorkflowStageBase):
    id: uuid.UUID
    workflow_id: uuid.UUID
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WorkflowTransitionBase(BaseModel):
    from_stage_id: uuid.UUID
    to_stage_id: uuid.UUID
    action: str
    required_permission: Optional[str] = None
    required_fields: Optional[List[str]] = []
    auto_create_task_type_id: Optional[uuid.UUID] = None
    assignment_mode: str = "automatic"
    active: bool = True


class WorkflowTransitionCreate(WorkflowTransitionBase):
    pass


class WorkflowTransitionOut(WorkflowTransitionBase):
    id: uuid.UUID
    workflow_id: uuid.UUID
    from_stage_name: Optional[str] = None
    to_stage_name: Optional[str] = None
    task_type_name: Optional[str] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class WorkflowBase(BaseModel):
    name: str
    code: str
    description: Optional[str] = None
    active: bool = True


class WorkflowCreate(WorkflowBase):
    pass


class WorkflowOut(WorkflowBase):
    id: uuid.UUID
    created_at: datetime
    updated_at: datetime
    stages: List[WorkflowStageOut] = []
    transitions: List[WorkflowTransitionOut] = []

    model_config = ConfigDict(from_attributes=True)
